tree/ntuple/v7/inc/ROOT/RNTupleFillContext.hxx

```diff
@@ -107,21 +111,36 @@ public:
    std::size_t Fill(REntry &entry)
    {
       RNTupleFillStatus status;
-      FillNoCommit(entry, status);
-      if (status.ShouldCommitCluster())
-         CommitCluster();
+      FillNoFlush(entry, status);
+      if (status.ShouldFlushCluster())
+         FlushCluster();
       return status.GetLastEntrySize();
    }
-   /// Ensure that the data from the so far seen Fill calls has been written to storage
-   void CommitCluster();
-
+   /// Flush column data, preparing for CommitCluster or to reduce memory usage. This will generally trigger compression
```

Comment:
It depends on what the page sink does. I think right now all of them compress in `CommitPage` (or at least spawn the task, in case of `RPageSinkBuf` + IMT), so I will update https://github.com/root-project/root/pull/16391 to remove the "generally".